### PR TITLE
sync to Neo.SmartContract.Framework v3.0.0-rc3

### LINKED ISSUE #46: Need new examples for rc2, pls.
Neo.Smartcontract.Framework v3.0.0-rc2 has changed a lot. All of the N3 examples out of date ...

### PR DESCRIPTION
**Pull request by superboyiii**
Close #46

### CODE REVIEW

**Comment by superboyiii** [comment]
@chenzhitong @shargon

**Comment by erikzhang** [comment]
Duplicate of #43

File: csharp/NEP17/NEP17.cs [hunk 1]
```diff
@@ -30,8 +34,12 @@ public partial class NEP17Demo : Framework.SmartContract
...
         public static bool Verify() => IsOwner();
 
-        public static string Symbol() => "TokenSymbol";
+        public static string Symbol() => "NTT";
 
         public static ulong Decimals() => 8;
+
+        public static UInt160 GetOwner() => Owner;
+
+        public static UInt160 GetOwner2() => Owner2;
```
Comment (chenzhitong) [inline csharp/NEP17/NEP17.cs:43]: For testing?

File: csharp/NEP17/NEP17.cs [hunk 2]
```diff
@@ -9,13 +9,17 @@ namespace Neo.SmartContract.Examples
     [ManifestExtra("Email", "dev@neo.org")]
     [ManifestExtra("Description", "This is a NEP17 example")]
     [SupportedStandards("NEP-17")]
-    [ContractPermission("*", "onNEP17Payment")]
+    [ContractPermission("*", "*")]
```
Comment (chenzhitong) [inline csharp/NEP17/NEP17.cs:12]: Why change this line

File: csharp/NEP17/NEP17.csproj [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
     <RootNamespace>Neo.SmartContract.Examples</RootNamespace>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Neo.SmartContract.Framework" Version="3.0.0-preview5" />
+    <PackageReference Include="Neo.SmartContract.Framework" Version="3.0.0-CI00300" />
```
Comment (chenzhitong) [inline csharp/NEP17/NEP17.csproj:9]: Miss Nuget.config
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear />
    <add key="MyGet-neo" value="https://www.myget.org/F/neo/api/v3/index.json" />
    <add key="NuGet.org" value="https://api.nuget.org/v3/index.json" />
  </packageSources>
</configuration>
```

**Comment by superboyiii** [comment]
> Duplicate of #43

We can iterate examples if any code style changes but not do anything together before release, it could help developers to know the changes step by step.

**Comment by satoshichou** [comment]
> > Duplicate of #43
> 
> We can iterate examples if any code style changes but not do anything together before release, it could help developers to know the changes step by step.

Agree, it's good for me. I'm wirting sc on neo and it makes the example more clear and these methods are all compatible with the latest compiler as what I tested.

**Review by chenzhitong — DISMISSED** [review]

**Review by shargon — APPROVED** [review]

**Comment by erikzhang** [comment]
The NEP-17 example should inherit `Nep17Token`.

**Comment by superboyiii** [comment]
> The NEP-17 example should inherit `Nep17Token`.
> Please update to 3.0.0-rc2
All done.

**Review by nicolegys — APPROVED** [review]
tested

File: csharp/NEP17/NEP17.cs [hunk 4]
```diff
@@ -10,28 +11,49 @@ namespace Neo.SmartContract.Examples
...
-        #region Token Settings
-        static readonly ulong MaxSupply = 10_000_000_000_000_000;
-        static readonly ulong InitialSupply = 2_000_000_000_000_000;
-        static readonly UInt160 Owner = "NiNmXL8FjEUEs1nfX9uHFBNaenxDHJtmuB".ToScriptHash();
-        static readonly ulong TokensPerNEO = 1_000_000_000;
-        static readonly ulong TokensPerGAS = 1;
-        #endregion
+        [InitialValue("NhGobEnuWX5rVdpnuZZAZExPoRs5J6D2Sb", ContractParameterType.Hash160)]
+        private static readonly UInt160 owner = default;
+        private static readonly byte[] ownerKey = "owner".ToByteArray();
```
Comment (erikzhang) [inline csharp/NEP17/NEP17.cs:18]: Why not use `byte`?

File: csharp/NEP17/NuGet.Config [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (erikzhang) [inline csharp/NEP17/NuGet.Config:1]: We don't need `NuGet.Config` if we are using rc2.

File: csharp/NEP17/NEP17.cs (hunk 4 above)
Comment (superboyiii) [reply csharp/NEP17/NEP17.cs:18]: > Why not use `byte`?

What's wrong with byte? You'd like ByteString?

Comment (erikzhang) [reply csharp/NEP17/NEP17.cs:18]: I think we should use `byte`.

**Comment by superboyiii** [comment]
@erikzhang pls review again.

File: csharp/NEP17/NEP17.cs [hunk 6]
```diff
@@ -10,28 +11,51 @@ namespace Neo.SmartContract.Examples
...
-        #region Token Settings
-        static readonly ulong MaxSupply = 10_000_000_000_000_000;
-        static readonly ulong InitialSupply = 2_000_000_000_000_000;
-        static readonly UInt160 Owner = "NiNmXL8FjEUEs1nfX9uHFBNaenxDHJtmuB".ToScriptHash();
-        static readonly ulong TokensPerNEO = 1_000_000_000;
-        static readonly ulong TokensPerGAS = 1;
-        #endregion
+        [InitialValue("NhGobEnuWX5rVdpnuZZAZExPoRs5J6D2Sb", ContractParameterType.Hash160)]
+        private static readonly UInt160 owner = default;
+        private static readonly byte[] ContractPrefix = new byte[] { 0x01, 0x01 };
```
Comment (shargon) [inline csharp/NEP17/NEP17.cs:18]: Only one byte?

File: csharp/NEP17/NEP17.cs [hunk 7]
```diff
@@ -10,28 +11,51 @@ namespace Neo.SmartContract.Examples
...
 
-        // When this contract address is included in the transaction signature,
-        // this method will be triggered as a VerificationTrigger to verify that the signature is correct.
-        // For example, this method needs to be called when withdrawing token from the contract.
-        public static bool Verify() => IsOwner();
+        public static UInt160 GetOwner()
+        {
+            return (UInt160)Storage.Get(Storage.CurrentContext, ownerKey);
+        }
+        public static new void Mint(UInt160 account, BigInteger amount)
```
Comment (shargon) [inline csharp/NEP17/NEP17.cs:35]:
```suggestion
        
        public static new void Mint(UInt160 account, BigInteger amount)
```

**Review by shargon — COMMENTED** [review]
Tested?

File: csharp/NEP17/NEP17.cs (hunk 6 above)
Comment (superboyiii) [reply csharp/NEP17/NEP17.cs:18]: > Only one byte?

How many bytes do you prefer?

Comment (shargon) [reply csharp/NEP17/NEP17.cs:18]: With only one we get the same result with less cost

**Review by nicolegys — APPROVED** [review]
Test pass

**Comment by superboyiii** [comment]
@shargon pls review again.

**Review by shargon — APPROVED** [review]

**Comment by superboyiii** [comment]
@erikzhang Merge?

File: csharp/NEP17/NEP17.cs (hunk 4 above)
Comment (erikzhang) [reply csharp/NEP17/NEP17.cs:20]: It is not solved.

Comment (superboyiii) [reply csharp/NEP17/NEP17.cs:21]: > It is not solved.

Done.

**Review by erikzhang — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Review by vncoelho — APPROVED** [review]
Working as expected. Thanks,

**Pull request merged**